src/python/pants/backend/go/util_rules/build_pkg.py

```diff
@@ -520,15 +527,25 @@ async def build_go_package(
     if symabis_path:
         compile_args.extend(["-symabis", symabis_path])
 
+    # If any assembly files are present, request the compiler write an "assembly header" with API metadata
+    # about the Go code that can be used by assembly files.
+    asm_header_path: str | None = None
+    if s_files:
+        obj_dir_path = PurePath(".", request.dir_path, "__obj__")
```

Comment:
Possibly worth using a magic constant `__obj__`? Unclear to me

Reply:
Yes but I'd consider it to be a separate refactor so the cgo code is included in that refactor.